include/aspect/material_model/rheology/peierls_creep.h

```diff
@@ -0,0 +1,136 @@
+       * Planetary Science Letters.
+       */
+      template <int dim>
+      class PeierlsCreep : public ::aspect::SimulatorAccess<dim>
+      {
+        public:
+          /**
+           * Constructor.
+           */
+          PeierlsCreep();
+
+          /**
+           * Declare the parameters this function takes through input files.
+           */
+          static
+          void
+          declare_parameters (ParameterHandler &prm);
+
+          /**
+           * Read the parameters from the parameter file.
+           */
+          void
+          parse_parameters (ParameterHandler &prm);
+
+          /**
+           * Compute the viscosity based on the selected Peierls creep flow law.
+           */
+          double
+          compute_viscosity (const double strain_rate,
+                             const double pressure,
+                             const double temperature,
+                             const unsigned int composition) const;
+
+        private:
+          /**
+           * Enumeration for selecting which type of Peierls creep flow law to use.
+           * Currently, the only available option directly calculates the viscosity
+           * using an approximation where the strain rate, rather than stress is used.
+           * This approximation requires specifying one fitting parameter (gamma) to
+           * obtain the best fit in the expected stress range for a given problem
+           * (gamma = stress / peierls_stress). The derivation for this approximation
+           * (derived by Magali Billen) can be found at:
+           * https://ucdavis.app.box.com/s/cl5mwhkjeabol4otrdukfcdwfvg9me4w/file/705438695737.
+           */
+          enum PeierlsCreepScheme
+          {
+            viscosity_approx
+          } peierls_creep_flow_law;
+
+          /**
+           * List of Peirls creep prefactors (A).
+           */
+          std::vector<double> prefactors;
+
+          /**
+           * List of Peierls creep stress exponents n.
+           */
+          std::vector<double> stress_exponents;
+
+          /**
+           * List of Peierls creep activation energies (E).
+           */
+          std::vector<double> activation_energies;
+
+          /**
+           * List of Peierls creep activation volumes V.
+           */
+          std::vector<double> activation_volumes;
+
+          /**
+           * List of Peierls stresses (sigma_p)
+           */
+          std::vector<double> peierls_stresses;
+
+          /**
+           * List of Peierls fitting parameters (gamma)
+           */
+          std::vector<double> fitting_parameters;
+
+          /**
+           * List of the first Peierls parameter related
+           * to dislocation glide (p)
+           */
+          std::vector<double> glide_parameters_p;
+
+          /**
+           * List of the second Peierls parameter related
+           * to dislocation glide (q)
```

Comment:
Since I am commenting anyway, please add periods at the end of documentation sentences above (or remove them uniformly, it is a matter of taste, but I am in the 'Comments are complete sentences too' fraction of the developer base :-)).